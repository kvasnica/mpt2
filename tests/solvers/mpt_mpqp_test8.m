function mpt_mpqp_test8

% tests that we properly display a warning when the mpQP is not positive
% definite:
%

M.H = -2;           % negative definite mpQP
M.F = 0; M.Y = 0; M.Cf = 2; M.Cx = 0; M.Cc = -1;
M.G = [-1; 1; 0; 0];
M.W = [5; 5; 0; 1];
M.E = [1; 1; 1; -1];
M.bndA = []; M.bndb = [];

fprintf('Expect warning that the mpQP is not positive definite:\n\n');
Pn = mpt_mpqp(M);

fprintf('\n\nExpect warning that the mpQP close to being negative definite:\n\n');
M.H = 1e-6;
Pn = mpt_mpqp(M);


fprintf('\n\nExpect warning that the mpQP lacks strict convexity:\n\n');
M.H = 0;
w = warning;
warning off
try
    Pn = mpt_mpqp(M);
end
warning(w);
