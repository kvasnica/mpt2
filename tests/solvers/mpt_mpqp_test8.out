Expect warning that the mpQP is not positive definite:

WARNING: Problem is not positive semidefinite! Your mpQP solution may be completely wrong.
mpt_mpqp: 1 regions


Expect warning that the mpQP close to being negative definite:

WARNING: QP is close to being (negative) semidefinite, may lead to troubles in mpQP solver.
mpt_mpqp:  Infeasible optimization problem from the begining


Expect warning that the mpQP lacks strict convexity:

WARNING: Lack of strict convexity may lead to troubles in mpQP solver.